services/vision/radiusclustering/radius_clustering.go

```diff
@@ -0,0 +1,58 @@
+// Package radiusclustering uses the 3D radius clustering algorithm as defined in the
+// RDK vision/segmentation package as vision model.
+package radiusclustering
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision/segmentation"
+)
+
+var model = resource.DefaultModelFamily.WithModel("radius_clustering_segmenter")
+
+func init() {
+	resource.RegisterService(vision.API, model, resource.Registration[vision.Service, *segmentation.RadiusClusteringConfig]{
+		DeprecatedRobotConstructor: func(ctx context.Context, r any, c resource.Config, logger golog.Logger) (vision.Service, error) {
+			attrs, err := resource.NativeConfig[*segmentation.RadiusClusteringConfig](c)
+			if err != nil {
+				return nil, err
+			}
+			actualR, err := utils.AssertType[robot.Robot](r)
+			if err != nil {
+				return nil, err
+			}
+			return registerRCSegmenter(ctx, c.ResourceName().Name, attrs, actualR)
+		},
+	})
+}
+
+// registerRCSegmenter creates a new 3D radius clustering segmenter from the config.
+func registerRCSegmenter(
+	ctx context.Context,
+	name string,
+	conf *segmentation.RadiusClusteringConfig,
+	r robot.Robot,
+) (vision.Service, error) {
+	_, span := trace.StartSpan(ctx, "service::vision::registerRadiusClustering")
+	defer span.End()
+	if conf == nil {
+		return nil, errors.New("config for radius clustering segmenter cannot be nil")
+	}
+	err := conf.CheckValid()
+	if err != nil {
+		return nil, errors.Wrap(err, "radius clustering segmenter config error")
+	}
+	segmenter := segmentation.Segmenter(conf.RadiusClustering)
```

Comment:
this would be nil to begin with if it were not a segmenter. seems like a ineffective cast?

Reply:
Segmenter is a function defintion (not an alias), so it needs to be cast to Segmenter in order to have the correct type

Reply:
can check b4

Reply:
changed the check to come before